gnovm/pkg/gnolang/realm.go

```diff
@@ -1435,6 +1435,7 @@ func fillTypesOfValue(store Store, val Value) Value {
 			fillTypesTV(store, &cur.Key)
 			fillTypesTV(store, &cur.Value)
 
+			fillValueTV(store, &cur.Key)
```

Comment:
Maybe we should make `ComputeMapKey` call `fillValueTV` when encountering RefValues, rather than doing it here?

For instance, I see that `*PointerType` doesn't correctly handle HeapItemValues or similar. `4280b` works because the case of `*StructType` in `ComputeMapKey` calls `fillValueTV`, but the case of `*PointerType` doesn't and I think it would fail...

Here I imagine something at the top of ComputeMapKey which checks if the value is a RefValue, and calls fillValueTV.